cmd/syslog/syslog.go

```diff
@@ -21,13 +24,27 @@ func main() {
 	server.SetHandler(handler)
 
 	if err := server.ListenUnixgram(listenAddress); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to listen to syslog unix socket %s, err %s", listenAddress, err)
 	}
 
 	if err := server.Boot(); err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to boot syslog server err %s", err)
 	}
 
+	// Unix sockets must be unlink()ed before being reused again.
+	// Handle common process-killing signals so we can gracefully shut down:
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, os.Kill, syscall.SIGTERM)
+	go func(c chan os.Signal) {
+		// Wait for a SIGINT or SIGKILL:
+		sig := <-c
+		log.Printf("Caught signal %s: shutting down.", sig)
+		// Stop listening (and unlink the socket if unix type):
+		_ = server.Kill()
+		_ = os.Remove(listenAddress)
```

Comment:
I'd record the error if remove fails. Otherwise you'll be back where you started.

Reply:
I was trying to make the signal handler very light weight